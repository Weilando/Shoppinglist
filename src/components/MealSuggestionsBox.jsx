import React from 'react';
import PropTypes from 'prop-types';
import { connect } from "react-redux";
import MealSuggestion from './MealSuggestion';

class MealSuggestionsBox extends React.Component {
  static propTypes = {
    suggestions: PropTypes.array,
  }

  static defaultProps = {
    suggestions: [],
  }

  getMealSuggestionList() {
    const suggestions = this.props.suggestions;
    const keys = [...Array(suggestions.length).keys()]; // Array with keys from 0 to entries.length

    return keys.map((tmpKey) =>
      <MealSuggestion
        key = {tmpKey}
        mealId = {Number(suggestions[tmpKey].idMeal)}
        mealName = {String(suggestions[tmpKey].strMeal)}
      />
    );
  }

  render() {
    let mealSuggestionList = this.getMealSuggestionList();

    return (
      <div className="InfoBox">
        <h3>Meal suggestions:</h3>
        <ul className="Shoppinglist">
          {mealSuggestionList}
        </ul>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    suggestions: state.mealSuggestions.mealSuggestionList,
  }
}

export default connect(mapStateToProps)(MealSuggestionsBox);
