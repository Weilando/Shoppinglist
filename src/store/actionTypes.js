export const ADD_ENTRY = 'ADD_ENTRY';
export const UPDATE_ENTRY = 'UPDATE_ENTRY';
export const DELETE_ENTRY = 'DELETE_ENTRY';
export const TOGGLE_STATUS = 'TOGGLE_STATUS';

export const UPDATE_MEAL_SUGGESTIONS = 'UPDATE_MEAL_SUGGESTIONS';
export const REMOVE_DUPLICATES = 'REMOVE_DUPLICATES';
