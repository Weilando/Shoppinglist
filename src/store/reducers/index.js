import { generalReducer } from './generalReducer';
import { mealSuggestionsReducer } from './mealSuggestionsReducer';
import { combineReducers } from 'redux';

// rootReducer
export default combineReducers({
    general: generalReducer,
    mealSuggestions: mealSuggestionsReducer
});
