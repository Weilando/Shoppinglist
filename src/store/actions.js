import { ADD_ENTRY, UPDATE_ENTRY, DELETE_ENTRY, TOGGLE_STATUS, UPDATE_MEAL_SUGGESTIONS, REMOVE_DUPLICATES } from './actionTypes';

let nextEntryId = 0;

export function addEntry(newContent) {
  return {
    type: ADD_ENTRY,
    payload: {
      id: ++nextEntryId,
      newContent
    }
  }
}

export function updateEntry(id, newContent) {
  return {
    type: UPDATE_ENTRY,
    payload: {
      id: id,
      newContent
    }
  }
}

export function deleteEntry(id) {
  return {
    type: DELETE_ENTRY,
    payload: { id: id }
  }
}

export function toggleStatus(id) {
  return {
    type: TOGGLE_STATUS,
    payload: { id: id }
  }
}

export function updateMealSuggestions(mealSuggestions) {
  return {
    type: UPDATE_MEAL_SUGGESTIONS,
    payload: { newMealSuggestions: mealSuggestions }
  }
}

export function removeDuplicates() {
  return {
    type: REMOVE_DUPLICATES
  }
}
