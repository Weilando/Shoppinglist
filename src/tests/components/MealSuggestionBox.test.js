import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/store';
import MealSuggestionsBox from '../../components/MealSuggestionsBox';

describe('snapshot tests for MealSuggestionsBox component', () => {
  it('should render empty MealSuggestionsBox component correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <MealSuggestionsBox />
      </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  })
});
