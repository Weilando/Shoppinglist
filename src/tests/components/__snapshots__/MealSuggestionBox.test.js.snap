// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot tests for MealSuggestionsBox component should render empty MealSuggestionsBox component correctly 1`] = `
<div
  className="InfoBox"
>
  <h3>
    Meal suggestions:
  </h3>
  <ul
    className="Shoppinglist"
  />
</div>
`;
